### PR TITLE
Add custom sections to systemd network

### PR DESCRIPTION
**Pull request by sallchr**
#### Pull Request (PR) description

Make it possible to define custom sections and settings via hash for systemd network interfaces

#### This Pull Request (PR) fixes the following issues

None, feature-request

### CODE REVIEW

File: manifests/provider/systemd.pp [hunk 1]
```diff
@@ -12,6 +12,7 @@
   Optional[String[1]] $description = undef,
   Optional[Integer[1200, 9000]] $mtu = undef,
   Array[Hash[String[1], Variant[String[1], Boolean]]] $routes = [],
+  Hash $sections = {},
```
Comment (TheMeier) [inline manifests/provider/systemd.pp:15]: Can this be a more specific custom type?

**Comment by TheMeier** [comment]
Not sure how you updated the REFERENCE.md but the validation is not valid. See https://voxpupuli.org/docs/how_to_run_tests/#referencemd-update

File: REFERENCE.md [hunk 2]
```diff
@@ -187,6 +187,26 @@ wireguard::interface { 'wg0':
 }
 ```
 
+##### Peer with one node, setup dualstack firewall rules and RoutingPolicyRule
```
Comment (TheMeier) [inline REFERENCE.md:190]: That should go into https://github.com/voxpupuli/puppet-wireguard/blob/master/manifests/interface.pp see, all the existing `@example` entries there, the generator will put it into the REFERENCE.md then

File: manifests/interface.pp [hunk 3]
```diff
@@ -112,6 +130,7 @@
   Optional[Integer[1200, 9000]] $mtu = undef,
   Optional[String[1]] $public_key = undef,
   Array[Hash[String[1], Variant[String[1], Boolean]]] $routes = [],
+  Hash[String, Hash[Any, String]] $sections = {},
```
Comment (smortex) [inline manifests/interface.pp:133]: According to your example, this seems to be:

```suggestion
  Hash[String, Hash[String, Any]] $sections = {},
```

IIRC, there is also a `Data` type that will only accept actual data and may be more suitable that `Any`.

File: manifests/provider/systemd.pp [hunk 4]
```diff
@@ -12,6 +12,7 @@
   Optional[String[1]] $description = undef,
   Optional[Integer[1200, 9000]] $mtu = undef,
   Array[Hash[String[1], Variant[String[1], Boolean]]] $routes = [],
+  Hash[String, Hash[Any, String]] $sections = {},
```
Comment (smortex) [inline manifests/provider/systemd.pp:15]: Same

File: manifests/interface.pp (hunk 3 above)
Comment (smortex) [reply manifests/interface.pp:133]: Also, some modules use a parameter name like `$custom_foo` or `$extra_foo` to add custom config.  Maybe somethig to think about before introducing this parameter.

Comment (sallchr) [reply manifests/interface.pp]: Thanks for your feedback. The pipeline is now successful.

**Comment by sallchr** [comment]
The last task would be to extend spec/defines/interface_spec.rb ?

**Pull request merged**